driver-core/src/main/java/com/datastax/driver/core/TokenRange.java

```diff
@@ -226,9 +247,8 @@ public boolean contains(Token token) {
     // * ]1,2] does not contain 1, but it contains the start of ]1,3]
     private boolean contains(Token token, boolean isStart) {
         boolean isAfterStart = isStart ? token.compareTo(start) >= 0 : token.compareTo(start) > 0;
-        boolean isBeforeEnd = end.equals(factory.minToken()) ||
-                (isStart ? token.compareTo(end) < 0 : token.compareTo(end) <= 0);
-        return isWrappedAround()
+        boolean isBeforeEnd = isStart ? token.compareTo(end) < 0 : token.compareTo(end) <= 0;
+        return isWrappedAround() || end.equals(factory.minToken())
```

Comment:
There is indeed a bug when `token.equals(minToken) && !isStart && this.end.equals(minToken)`. I'm opening a separate ticket to keep track of it.